esmvalcore/_task.py

```diff
@@ -711,19 +714,59 @@ def get_independent(self) -> 'TaskSet':
                 independent_tasks.add(task)
         return independent_tasks
 
-    def run(self, max_parallel_tasks: int = None) -> None:
+    def run(self, cfg) -> None:
         """Run tasks.
 
         Parameters
         ----------
-        max_parallel_tasks : int
-            Number of processes to run. If `1`, run the tasks sequentially.
+        cfg : dict
+            Config-user dict.
         """
-        if max_parallel_tasks == 1:
+        max_parallel_tasks = cfg['max_parallel_tasks']
+        if max_parallel_tasks == -1:
+            self._run_dask(cfg)
+        elif max_parallel_tasks == 1:
             self._run_sequential()
         else:
             self._run_parallel(max_parallel_tasks)
 
+    def _run_dask(self, cfg) -> None:
+        """Run tasks using dask."""
+        # Configure dask
+        client_args = cfg.get('dask', {}).get('client', {})
+        cluster_args = cfg.get('dask', {}).get('cluster', {})
+        cluster_type = cluster_args.pop(
+            'type',
+            'dask.distributed.LocalCluster',
+        )
+        cluster_scale = cluster_args.pop('scale', 1)
+
+        # STart cluster
+        cluster_module_name, cluster_cls_name = cluster_type.rsplit('.', 1)
+        cluster_module = importlib.import_module(cluster_module_name)
+        cluster_cls = getattr(cluster_module, cluster_cls_name)
+        cluster = cluster_cls(**cluster_args)
+        cluster.scale(cluster_scale)
+
+        # Connect client and run computation
+        with dask.distributed.Client(cluster, **client_args) as client:
```

Comment:
Hey @bouweandela, just a minor suggestion here. 

Currently, a cluster is instantiated even if an external cluster address is passed from input, and this should solve this. Also, I believe that Dask clusters have `n_workers` as input argument, which makes `cluster_scale` redundant. 

```suggestion
        # Configure dask
        client_args = cfg.get('dask', {}).copy()
        cluster_args = client_args.pop('cluster', {})

        if cluster_args:
            # Start cluster
            cluster_type = cluster_args.pop(
                'type',
                'dask.distributed.LocalCluster',
            )
            cluster_module_name, cluster_cls_name = cluster_type.rsplit('.', 1)
            cluster_module = importlib.import_module(cluster_module_name)
            cluster_cls = getattr(cluster_module, cluster_cls_name)
            cluster = cluster_cls(**cluster_args)
            client_args["address"] = cluster

        # Connect client and run computation
        with dask.distributed.Client(**client_args) as client:
```

Reply:
Thanks, I improved the code! Could you have a look if it works correctly now?